### PR TITLE
(PC-14455) feat(PhoneValidation): display Stepper version based on backend value

### PR DESCRIPTION
**Pull request by yorickeando**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-14455

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

#### With FF on:

Capture d’écran 2022-05-23 à 16 18 32

#### With FF off:

Capture d’écran 2022-05-23 à 16 19 00

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/navigation/RootNavigator/types.ts [hunk 1]
```diff
@@ -24,6 +24,9 @@ export type IdentityCheckRootStackParamList = {
   NavigationIdentityCheck: undefined
   // Stepper
   IdentityCheckStepper: undefined
+  // PhoneValidation
+  // TODO : replace with SetPhoneNumberV2 once it's designed and usable in development
```
Comment (anoukhello) [inline src/features/navigation/RootNavigator/types.ts:28]: Un ptit numéro de ticket ?

File: src/features/profile/components/NonBeneficiaryHeader.web.test.tsx [hunk 2]
```diff
@@ -176,6 +180,7 @@ describe('<NonBeneficiaryHeader/>', () => {
       mockNextSubscriptionStep = {
         allowedIdentityCheckMethods: [IdentityCheckMethod.ubble],
         nextSubscriptionStep: null,
+        stepperIncludesPhoneValidation: false,
```
Comment (anoukhello) [inline src/features/profile/components/NonBeneficiaryHeader.web.test.tsx:183]: Ça fait beaucoup de mock à modifier, tu penses qu'on pourrait faire des fixtures ? Ou chaque test a besoin d'une réponse bien spécifique ?

Comment (yorickeando) [reply src/features/profile/components/NonBeneficiaryHeader.web.test.tsx:183]: Bonne suggestion. Je vois si je fais ça maintenant ou bien si je me fais un petit ticket d'amélio.

File: src/features/identityCheck/__test__/useIdentityCheckSteps.test.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  })
+  it('should include IdentityCheckSchoolType if reducer has school types', () => {
+    mockIdentityCheckState = {
+      ...mockState,
+      profile: { ...mockState.profile, hasSchoolTypes: true },
+    }
+    const steps = useIdentityCheckSteps()
+    expect(steps[0].screens.includes('IdentityCheckSchoolType')).toEqual(true)
+  })
+  it('should not include IdentityCheckSchoolType if reducer has school types', () => {
```
[the remark this answers is not in the document]
Comment (yorickeando) [reply src/features/identityCheck/__test__/useIdentityCheckSteps.test.tsx]: vu !

**Pull request merged**